### ISSUE TITLE
Follow up on Offhand Swing Timers

### ISSUE THREAD

**Issue by Manpower99** [gamefreedomgit/Frostmourne#1212]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (https://www.wowhead.com/wotlk)

**Description:**
https://github.com/gamefreedomgit/Frostmourne/issues/302
Have to remake it with better explanation since this is affecting all dual wielding classes/specs and has not been adressed after so long.

So currently multiple things seem to be incorrect
1.Triggering an attack From further than melee range does not seem to put offhand's swing at 50% if it is further than 50% total  progress.
And triggering an attack within melee range should sync them both to hit at the same time.

2.When swapping target the offhand does not seem to be set back to 50% of the swing's total progress if it is at or further than 50% total progress.
using macros such as : 
/cleartarget
/targetlasttarget
/startattack
or
/tar Player
/targetlasttarget
/startattack

those two should basically do the same as swapping target without actually doing so and thus desync the Offhand back to 50% if it is at 50% or further than 50% total progress.
Info straight from classic wotlk
Pressing this macro will set your off-hand swing back to 50% completion. If your OH swing is below 50% of completion, it won't do anything. If it's at 60%, it will set the swing back by 10%. If it's at 99%, it will set the swing back by 49%. The later into the swing your press the desync macro, the better you will stagger your weapons for steadier incoming rage.

Reasons why these issues should be fixed : 
For Warriors the desync improves rage generation, and prevent munching of Deep Wounds
For Enhancement Shaman can increase there Flurry uptime by consuming only one charge by MH and OH swing (this quirk of enhancement shaman is already fixed but if you fix the issues im reporting it will improve the gameplay).

there are many videos explaining how these things work even just visually by looking at a swingtimer.
https://www.youtube.com/watch?v=JJ5HmgZOZYA
In this video at 2:00 this player is stationary and constantly attacking does not use the macro, but because the first Offhand was delayed and thus never Synced and never need to be Desynced

https://www.youtube.com/watch?v=ETsvFN5GD8k
is perhaps one of the best videos for showing/explaining it.

https://www.youtube.com/watch?v=In78lWEWyH8
has a perfect explanation for how it should work for Warriors

I have more videos and more proof if required.

Tldr: Mainhand Attack swing progress shouldnt be touched as it is working fine
Offhand attack swing progress should be set back to 50% progress if
~~1.you trigger the attack from further than melee range while your offhand progress is above 50%.~~
2.you swap target while your offhand progress is above 50%.
3.you use a macro such as the ones i linked above WHILE your offhand progress is above 50%

1.is incorrect, it should actually Sync both weapons at 100% if you do right click or /startattack on the target while further than melee range, this of course would only happen if both swings are already at 100% progress but didnt hit yet or it is the start of combat

First attack while already in melee range should however put both weapons at a synced attack progress from the start.

To properly conduct this test to see for yourself equip two weapons of equal base attack speed such as x2 Havocs for Enha Shaman or a double 3.6 Gloranzelg so the weapons dont naturally desync themselves by being of different base attack speeds.

**How to reproduce:**

**How it should work:**

**Database links:**

**Comment by Manpower99** [comment]
> `1.Triggering an attack From further than melee range does not seem to put offhand's swing at 50% if it is further than 50% total progress.` So if this is further than melee range - how to make a swing from offhand?
> 
> Also - 50% of offhand progress. Haste does count into it? E.g. your base weapon spees id 3.6, but after all haste modifiers your offhand attack timer is let's say 2.9 What humber should we take into math?


Haste should not count to it no, its simply setting it at 50% of the total swing timer regardless. (to all of the mentioned effects)

As for the 1. its about triggering it ( /startattack or right clicking it while out of melee range )
which made me realise i made a mistake in my bug report here, because as i scoured more into the wotlk classic discords and rewatched more video i understood that the 1. is actually incorrect (on my part).
Basically triggering an autoattack by rightclicking on target or /startattack while further than melee range actually should sync your weapons at 100%.
in this timestamped [video ](https://youtu.be/In78lWEWyH8?si=GDys0ZD6qI6r7l3l&t=250)he started the attack from further than melee range and when his attack finally landed you can see they were in fact synced. this should be the case for both when charging and whenever you start the swing and then walk into melee range both being the same thing.

**Comment by UltraNix** [comment]
Fixed.

**Comment by Manpower99** [comment]
> Fixed.

https://cdn.discordapp.com/attachments/1267276893602971701/1270873728762384505/2024-08-08_00-34-10.mp4?ex=66b7ebe4&is=66b69a64&hm=0f2f75305e45f604801b8602729d514ed929e535fae8fb8fa545c9165e31147f&

It seems to break often especially with sun lute, works with some weapons and others not.
